File: frontend/src/types/patient.ts
export interface Patient {
  id: string;
  hospital_id: string;
  patient_reference_number: string;
  first_name: string;
  last_name: string;
  date_of_birth: string;
  gender: string;
  blood_group: string | null;
  phone_country_code: string;
  phone_number: string;
  email: string | null;
  address_line_1: string;
  address_line_2: string | null;
  city: string | null;
  state: string | null;
  pin_code: string | null;
  country: string | null;
  age_years: number | null;
  age_months: number | null;
  marital_status: string | null;
  emergency_contact_name: string | null;
  emergency_contact_phone: string | null;
  emergency_contact_relation: string | null;
  photo_url: string | null;
  known_allergies: string | null;
  chronic_conditions: string | null;
  is_deleted: boolean;
  created_by: string | null;
  updated_by: string | null;
  created_at: string;
  updated_at: string;
}

export interface PatientListItem {
  id: string;
  patient_reference_number: string;
  first_name: string;
  last_name: string;
  gender: string;
  phone_country_code: string;
  phone_number: string;
  email: string | null;
  city: string | null;
  blood_group: string | null;
  created_at: string;
}

export interface PaginatedResponse<T> {
  total: number;
  page: number;
  limit: number;
  total_pages: number;
  data: T[];
}

export interface PatientCreateData {
  first_name: string;
  last_name: string;
  date_of_birth: string;
  gender: string;
  blood_group?: string;
  phone_country_code: string;
  phone_number: string;
  email?: string;
  address_line_1: string;
  address_line_2?: string;
  city?: string;
  state?: string;
  pin_code?: string;
  country?: string;
  age_years?: number;
  age_months?: number;
  marital_status?: string;
  emergency_contact_name?: string;
  emergency_contact_phone?: string;
  emergency_contact_relation?: string;
}

File: backend/app/models/patient.py
"""
Patient model — matches new hms_db UUID schema.
"""
import uuid
from sqlalchemy import (
    Column, String, Boolean, DateTime, Date, Integer, Text, ForeignKey, UniqueConstraint
)
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..database import Base


class Patient(Base):
    __tablename__ = "patients"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    hospital_id = Column(UUID(as_uuid=True), ForeignKey("hospitals.id"), nullable=False)
    patient_reference_number = Column(String(12), nullable=False, index=True)
    title = Column(String(10))
    first_name = Column(String(100), nullable=False)
    last_name = Column(String(100), nullable=False)
    date_of_birth = Column(Date)
    age_years = Column(Integer)
    age_months = Column(Integer)
    gender = Column(String(20), nullable=False)
    blood_group = Column(String(5))
    marital_status = Column(String(20))
    phone_country_code = Column(String(5), nullable=False, default="+1")
    phone_number = Column(String(15), nullable=False)
    secondary_phone = Column(String(20))
    email = Column(String(255))
    national_id_type = Column(String(30))
    national_id_number = Column(String(50))
    address_line_1 = Column(String(255))
    address_line_2 = Column(String(255))
    city = Column(String(100))
    state_province = Column(String(100))
    postal_code = Column(String(20))
    country = Column(String(100), default="India")
    photo_url = Column(String(500))
    emergency_contact_name = Column(String(200))
    emergency_contact_phone = Column(String(20))
    emergency_contact_relation = Column(String(50))
    known_allergies = Column(Text)
    chronic_conditions = Column(Text)
    notes = Column(Text)
    preferred_language = Column(String(10), default="en")
    is_active = Column(Boolean, default=True)
    registered_at = Column(DateTime(timezone=True), server_default=func.now())
    created_at = Column(DateTime(timezone=True), server_default=func.now(), index=True)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    created_by = Column(UUID(as_uuid=True), ForeignKey("users.id"))
    updated_by = Column(UUID(as_uuid=True), ForeignKey("users.id"))
    is_deleted = Column(Boolean, default=False)
    deleted_at = Column(DateTime(timezone=True))

    # Relationships
    hospital = relationship("Hospital", foreign_keys=[hospital_id])

    __table_args__ = (
        UniqueConstraint("hospital_id", "patient_reference_number", name="uq_patient_prn_hospital"),
    )

    @property
    def full_name(self) -> str:
        parts = []
        if self.title:
            parts.append(self.title)
        parts.append(self.first_name)
        parts.append(self.last_name)
        return " ".join(parts)

